recipes/mariadb-connector-cpp/all/conanfile.py

```diff
@@ -0,0 +1,135 @@
+            raise ConanInvalidConfiguration("gnutls not yet available in CCI")
+
+    def build_requirements(self):
+        self.tool_requires("cmake/[>=3.23 <4]")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+        apply_conandata_patches(self)
+
+    def generate(self):
+        deps = CMakeDeps(self)
+        deps.generate()
+
+        tc = CMakeToolchain(self)
+        tc.cache_variables["CMAKE_PROJECT_mariadb_connector_cpp_INCLUDE"] = os.path.join(self.source_folder, "conan_cmake_project_include.cmake")
+        tc.cache_variables["WITH_UNIT_TESTS"] = False
+        tc.cache_variables["INSTALL_BINDIR"] = "bin"
+        tc.cache_variables["INSTALL_LIBDIR"] = "lib"
+        tc.cache_variables["INSTALL_PLUGINDIR"] = os.path.join("lib", "plugin").replace("\\", "/")
+        tc.cache_variables["USE_SYSTEM_INSTALLED_LIB"] = True
+        tc.cache_variables["MARIADB_LINK_DYNAMIC"] = True
+
+        if self.settings.os == "Windows":
+            tc.cache_variables["CONC_WITH_MSI"] = False
+            tc.cache_variables["WITH_MSI"] = False
+
+        # To install relocatable shared libs on Macos
+        tc.cache_variables["CMAKE_POLICY_DEFAULT_CMP0042"] = "NEW"
+
+        tc.generate()
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def package(self):
+        cmake = CMake(self)
+        cmake.install(component="Headers")
+        if self.options.shared:
+            cmake.install(component="SharedLibraries")
+        else:
+            cmake.install(component="Development")
+
+        copy(self, "COPYING", src=os.path.join(self.source_folder), dst=os.path.join(self.package_folder, "licenses"))
+
+        fix_apple_shared_install_name(self)
+
+    def package_info(self):
+        self.cpp_info.set_property("cmake_find_mode", "both")
+        self.cpp_info.set_property("cmake_file_name", "mariadb-connector-cpp")
+        self.cpp_info.set_property("cmake_target_name", "mariadb-connector-cpp::mariadb-connector-cpp")
+        self.cpp_info.set_property("pkg_config_name", "libmariadbcpp")
+
+        if self.settings.os in ["Linux", "FreeBSD"]:
+            self.cpp_info.system_libs = ["dl", "m", "pthread"]
+        elif self.settings.os == "Windows":
+            self.cpp_info.system_libs = ["ws2_32", "shlwapi"]
+            if self.options.with_ssl == "schannel":
+                self.cpp_info.system_libs.append("secur32")
+
+        if self.options.shared:
+            self.cpp_info.libs = ["mariadbcpp"]
+            if self.settings.os == "Windows":
+                self.cpp_info.defines.append("MARIADB_EXPORTED=__declspec(dllimport)")
+        else:
+            if self.settings.os == "Windows":
+                self.cpp_info.defines.append("MARIADB_STATIC_LINK")
+                self.cpp_info.libs = ["mariadbcpp-static"]
+            else:
+                self.cpp_info.libs = ["mariadbcpp"]
```

Comment:
I've one more additional question. Upstream is generating two static libraries, one pointing to another:
`libmariadbcpp-static.a` and `libmariadbcpp.a`. 
Why are we keeping both and not just simply deleting one of them and why does it seems to be needed the `-static.a` one on windows?

Reply:
On windows, libmariadbcpp is not created as a link to libmariadbcpp-static.

Reply:
we can just rename it if you want

Reply:
my bad, libmariadbcpp-static is not created on macos, but on windows, it's created

Reply:
> On windows, libmariadbcpp is not created as a link to libmariadbcpp-static.

it is true